src/builtins/BuiltinAtomics.cpp

```diff
@@ -139,46 +139,26 @@ static Value getModifySetValueInBuffer(ExecutionState& state, ArrayBuffer* buffe
     ASSERT(indexedPosition + elemSize <= buffer->byteLength());
     uint8_t* rawStart = const_cast<uint8_t*>(buffer->data()) + indexedPosition;
 
-    if (v2.isInt32()) {
-        switch (type) {
-        case TypedArrayType::Int8:
-            return Value(atomicOperation<int8_t>(rawStart, v2.asInt32(), op));
```

Comment:
Since `value` is converted to the large type with same signed format, it doesn't make any problem I think.
Also, this `int64_t` typed value is sequentially converted to the same type `T` again for `__atomic` builtin functions.